### ISSUE TITLE
login link and how to control the redirect back from idp

### ISSUE THREAD

**Issue by brianwcook** [pingidentity/mod_auth_openidc#137]
Hello,
I have a slightly weird use case I suppose that I am trying to implement.

I have a page that is not protected but I want to display a link with embedded access code if the person is logged in to the IdP.   I am trying to figure out how to build a login href that will send the user through the auth code flow process and then return them to the original page.  then I plan to use the xmlhttprequest to get the access tokens from the header.  If the access_token is not in the header then I display the login link.

I can setup a href to /callback but it complains that I should not send someone there directly.  makes sense, but how do I send them to a different protected page and then have them come back to the unprotected one?
 
I have this in httpd conf

OIDCProviderMetadataURL https://<snipped>/auth/realms/openshift/.well-known/openid-configuration
OIDCClientID <snipped>
OIDCClientSecret <snipped>
OIDCRedirectURI http://[id addr]/OIDC/redirect_uri
OIDCCryptoPassphrase <snipped>
  <Location "/">     <--- this is where my unprotected page is (index.conf)

  </Location>

  <Location "/OIDC">
    AuthName "Registry Authentication"
    Authtype openid-connect
    Require valid-user
    LogLevel debug
  </Location>
</VirtualHost>

Thanks,
Brian

**Comment by zandbelt** [comment]
sound like you could use the so-called 3rd-party initiated SSO feature in the href that is described here near the end of the section https://github.com/pingidentity/mod_auth_openidc#openid-connect-sso-with-multiple-openid-connect-providers:

so it would look like:

    <oidc_callback>?target_link_uri=<target_link_uri>&iss=<issuer>

you may also be interested to use extra parameters:

    auth_request_params=<urlencoded-query-string>]

and provide "prompt=none" as part of those

**Comment by brianwcook** [comment]
I guess I am not understanding the flow of this.  Do I have to setup the metadata directories and everything described to make this work?  I added the ?target_link_uri= parameter to my login href that sends the user to a blank, protected page but after auth I was just left on the blank page and not redirected.

**Comment by brianwcook** [comment]
Okay, I got this working.  I had to setup the metadata directory and remove the static config from the httpd.conf and then I constructed the login link as you stated about to the callback URL.  Prompt=none did not work, it just gives a message saying you are not logged in and prevents redirection to the IdP login page.  Removing it made it work as I expected it to.  

One thing I had to do which might be helpful to someone is create a location for the unprotected page:

`
<Location "/">
        AuthName "auth"
        Authtype openid-connect
        Require all granted
  </Location>`

without doing that, trying to get the access_token passed in the header didn't work.

Thanks
Brian
